### PR TITLE
[RAC-4344]Default docker-compose.yml remove ELK

### PR DESCRIPTION
**Pull request by HQebupt**
By default, the docker-compose.yml carries elasticsearch/logstash/kibana
It's very convenient for advanced user who wants ELK integration with RackHD . but it's also an overhead to new user who don't need them. That's why there was the docker-compose-mini.yml. (From RAC-4344 @panpan0000 )

So we want to make the docker-compose-mini.yml as a default one in README.MD.
Related PR: [[RAC-4344]remove the docker-compose-mini.yml #279](https://github.com/RackHD/on-build-config/pull/279)
@changev @panpan0000 

**Summary:**
We make the docker-compose-mini.yml as default, and rename it with docker-compose.yml. For building image, we add two modules (on-core, on-tasks) into docker-compose.yml.
Syncly, we modify wherever the docker-compose-mini.yml exits in CI/CD code. And we also fix the README.md & bash scripts.
Besides, the docker-compose-mini-dev.yml is useless, we delete it.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -112,12 +112,12 @@ Now start `pxe-1` from VirtualBox. You should see it boot and auto automatically
     * By default the Vagrant Docker VM exposes all related ports. Some of which are only necessary for development and debugging. You can disable any ports you do not wish to use, or change the which port on the host they map too.
     * Ensure you have the right version of Vagrant for VirtualBox. Later versions of VirtualBox require a more recent version of Vagrant.
 
-## Run a minimal set of RackHD containers (without ELK).
-For who doesn't need ELK running along with RackHD, due to system performance restriction or docker-pull network bandwidth restriction: an alternative docker-compose file can be used by "-f" option:
+## Run RackHD containers (with ELK).
+For who need ELK running along with RackHD, due to system performance restriction or docker-pull network bandwidth restriction: an alternative docker-compose file can be used by "-f" option:
```
Comment (panpan0000) [inline docker/README.md]: BTW: Can you also teach people how to get touch with ELK from web browser ?

File: docker/docker-compose-elk.yml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    volumes:
+      - "./files/mount:/RackHD/files"
+
+  # test:
+  #   build: "../test"
+  #   image: "rackhd/test"
+  #   network_mode: "host"
+  #   privileged: true
+
+  # RackHD services
```
Comment (iceiilin) [inline docker/docker-compose-elk.yml]: can theses comments be removed?

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -142,50 +113,3 @@ services:
...
-      - logstash
-      - mongo
-      - rabbitmq
-      - statsd
-      - syslog
-    image: rackhd/on-wss:${TAG}
-    network_mode: "host"
-    privileged: true
-    volumes:
-      - "./monorail:/opt/monorail"
```
Comment (iceiilin) [inline docker/docker-compose.yml]: on-wss is supposed to be used by on-web-ui inside on-http, and suggest not to remove here.

File: docker/docker-compose.yml [hunk 4]
```diff
@@ -43,24 +43,17 @@ services:
...
-    command: "/bin/echo \"exit\""
-
-  tasks:
-    build: "../on-tasks"
-    image: rackhd/on-tasks:${TAG}
-    command: "/bin/echo \"exit\""
+  #  core:
+  #    build: "../on-core"
+  #    image: rackhd/on-core:latest
+  #    command: "/bin/echo \"exit\""
```
Comment (iceiilin) [inline docker/docker-compose.yml]: Why removing these two here while keeping them in docker-compose-elk.yml?

File: docker/docker-compose-elk.yml (hunk 2 above)
Comment (HQebupt) [inline docker/docker-compose-elk.yml]: OK, it should be removed.

File: docker/docker-compose.yml (hunk 4 above)
Comment (HQebupt) [inline docker/docker-compose.yml]: The docker-compose-elk.yml is copied from the old file: docker-compose.yml. I have fix it done.

File: docker/docker-compose.yml (hunk 3 above)
Comment (HQebupt) [inline docker/docker-compose.yml]: Yeah, agree.

File: docker/README.md (hunk 1 above)
Comment (HQebupt) [inline docker/README.md]: OK, I add one line in README.md. "Open http://127.0.0.1:5601 in a web browser, you can see kibana's website."

**Comment by HQebupt** [comment]
docker-compose-mini.yml is used by CI/CD. so it should not be removed.  And docker-compose-mini-dev.yml isn't used, It can be removed. What's your ideas? @changev @panpan0000 @iceiilin

**Comment by panpan0000** [comment]
Looks good.
Please summary all of your changes briefly in the PR Description

File: docker/docker-compose-elk.yml [hunk 5]
```diff
@@ -66,21 +66,38 @@ services:
     build: "../on-http"
     depends_on:
       - files
+      - logstash
       - mongo
       - rabbitmq
+      - statsd
+      - tasks
```
Comment (changev) [inline docker/docker-compose-elk.yml]: on-http truely depends on tasks, but the denpendency has been defined in Dockerfile of on-http.

If indicate dependency here then the http will wait for tasks running container, and IMO this is not expected here.

Suggestion: remove tasks.

File: docker/docker-compose-elk.yml [hunk 6]
```diff
@@ -91,9 +108,12 @@ services:
     build: "../on-taskgraph"
     depends_on:
       - dhcp
+      - logstash
       - mongo
       - rabbitmq
       - syslog
+      - statsd
+      - tasks
```
Comment (changev) [inline docker/docker-compose-elk.yml]: same as above

File: docker/docker-compose-elk.yml (hunk 5 above)
Comment (HQebupt) [inline docker/docker-compose-elk.yml]: It makes sense. Thanks @changev

**Pull request merged**